src/SprykerEco/Zed/Braintree/Business/Payment/Transaction/CaptureItemsTransaction.php

```diff
@@ -182,16 +182,16 @@ protected function getDecimalAmountValueFromInt(int $amount): float
      */
     protected function captureShipmentAmount(): void
     {
-        $orderTransfer = $this->salesFacade->getOrderByIdSalesOrder($this->transactionMetaTransfer->getIdSalesOrder());
-        $braintreePayment = $this->braintreeRepository->findPaymentBraintreeBySalesOrderId($orderTransfer->getIdSalesOrder());
+        $orderTransfer = $this->salesFacade->getOrderByIdSalesOrder($this->transactionMetaTransfer->getIdSalesOrderOrFail());
+        $braintreePayment = $this->braintreeRepository->findPaymentBraintreeBySalesOrderId($orderTransfer->getIdSalesOrderOrFail());
         $amount = $this->getShipmentExpenses($orderTransfer->getExpenses());
 
         if (!$braintreePayment || $braintreePayment->getIsShipmentPaid() || $amount === 0) {
             return;
         }
 
         $shipmentTransactionMetaTransfer = clone $this->transactionMetaTransfer;
-        $shipmentTransactionMetaTransfer->setCaptureShipmentAmount($this->getDecimalAmountValueFromInt($amount));
+        $shipmentTransactionMetaTransfer->setCaptureShipmentAmount((int)$this->getDecimalAmountValueFromInt((int)$amount));
```

Comment:
amount is already `int|null` after `this->getShipmentExpenses($orderTransfer->getExpenses());` call

add this check above if needed instead of casting types
higher you have early return in case `$amount === 0` 
what is the reason to skipp possible `null` and cast in here to `0`?

Reply:
Fixed by skipping nulls too.